libtiledbsoma/src/soma/managed_query.h

```diff
@@ -167,10 +167,26 @@ class ManagedQuery {
     }
 
     /**
-     * @brief Submit the query.
+     * @brief Set column data for write query.
      *
+     * @param column_buffer Column data
      */
-    void submit();
+    template <typename T>
+    void set_column_data(std::string column_name, std::vector<T>& buf) {
+        query_->set_data_buffer(column_name, buf);
+    }
+
+    /**
+     * @brief Submit the read query.
+     *
+     */
+    void submit_read();
+
+    /**
+     * @brief Submit the write query.
+     *
+     */
+    void submit_write();
```

Comment:
Ideally the read and write queries would not be coupled in the same file. Because writes are one-shot (non-global-order at least but we don't do these) I would like to see a single `write` function that accepts all buffers and does the entire operation, without requiring its consumers to handle query objects.

Reply:
> I would like to see a single write function that accepts all buffers and does the entire operation, without requiring its consumers to handle query objects.

I agree this would be best. For now though, I'm not sure of a simple way to have a single write function that can deal with taking in buffers of different datatypes. It'll be straightforward though once we switch over to using Arrow structs.